cli/common.go

```diff
@@ -61,6 +61,14 @@ func GenerateBuildResultForCommand(cmd *cli.Command) (*core.BuildResult, *a.App,
 	if err != nil {
 		return nil, nil, nil, fmt.Errorf("error creating env: %w", err)
 	}
+	// Make the build platform available to the build plan
+	if env.Variables["TARGETPLATFORM"] == "" {
+		platform, err := getPlatform(cmd.String("platform"))
+		if err != nil {
+			return nil, nil, nil, cli.Exit(err, 1)
+		}
+		env.Variables["TARGETPLATFORM"] = platform.String()
+	}
```

Comment:
Thoughts on passing it in the options and then having on the generate context? This works too, but I think it might be a bit confusing having `os.Getenv` in the providers

Reply:
Yeah I can put in there. I was between those two options.

Reply:
Actually I think we can maybe remove this entirely if (again) we drop musl